src/core/vector/qgsvectorlayertools.cpp

```diff
@@ -69,34 +76,25 @@ bool QgsVectorLayerTools::copyMoveFeatures( QgsVectorLayer *layer, QgsFeatureReq
     else
     {
       newFeature = QgsVectorLayerUtils::createFeature( layer, f.geometry(), f.attributes().toMap() );
+      if ( !layer->addFeature( newFeature ) )
+      {
+        couldNotWriteCount++;
+        QgsDebugError( QStringLiteral( "Could not add new feature. Original copied feature id: %1" ).arg( f.id() ) );
+      }
     }
 
     // translate
     if ( newFeature.hasGeometry() )
     {
       QgsGeometry geom = newFeature.geometry();
-      geom.translate( dx, dy );
-      newFeature.setGeometry( geom );
-#ifdef QGISDEBUG
-      const QgsFeatureId fid = newFeature.id();
-#endif
-      // paste feature
-      if ( !layer->addFeature( newFeature ) )
-      {
-        couldNotWriteCount++;
-        QgsDebugError( QStringLiteral( "Could not add new feature. Original copied feature id: %1" ).arg( fid ) );
-      }
-      else
+      fidList.insert( newFeature.id() );
```

Comment:
We should not do this part if the addFeature has failed, no ? 

It looks like also the `fidList` param is usefull only for `request` parameter which is never used. Could we get rid of these?

Reply:
Fixed, thanks for spotting this.

The request parameter is passed by reference in the function, I assume the intent was for the request to have filter IDs that match the created features. In any case, I'll avoid changing for now since I wasn't the original author.